server/src/main/java/io/crate/replication/logical/SynchronizeTableDefinitionsTask.java

```diff
@@ -0,0 +1,212 @@
+    private final Iterator<TimeValue> delay;
+
+    private Scheduler.Cancellable cancellable;
+    private boolean isStarted = false;
+
+    public SynchronizeTableDefinitionsTask(Settings settings, ThreadPool threadPool, Function<String, Client> remoteClient, ClusterService clusterService) {
+        this.threadPool = threadPool;
+        this.remoteClient = remoteClient;
+        this.clusterService = clusterService;
+        this.pollDelay = LogicalReplicationSettings.REPLICATION_READ_POLL_DURATION.get(settings);
+        this.delay = BackoffPolicy.exponentialBackoff(pollDelay, 8).iterator();
+    }
+
+    public void start() {
+        assert isStarted == false : "SynchronizeTableDefinitionsTask is already started";
+        assert clusterService.state().getNodes().getLocalNode().isMasterNode() : "SynchronizeTableDefinitionsTask must only be executed on the master node";
+        var executor = threadPool.executor(ThreadPool.Names.LOGICAL_REPLICATION);
+
+        try {
+            executor.execute(this::run);
+        } catch (EsRejectedExecutionException e) {
+            threadPool.schedule(
+                this::run,
+                delay.next(),
+                ThreadPool.Names.LOGICAL_REPLICATION
+            );
+        }
+        isStarted = true;
+    }
+
+    public boolean addRemoteClusterToTrack(String remoteCluster) {
+        return subscriptionsToTrack.add(remoteCluster);
+    }
+
+    public boolean isStarted() {
+        return isStarted;
+    }
+
+    private void run() {
+        for (String subscriptionName : subscriptionsToTrack) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Start syncing table definitions for subscription {}", subscriptionName);
+            }
+            getRemoteClusterState(subscriptionName, new ActionListener<>() {
+                @Override
+                public void onResponse(ClusterState remoteClusterState) {
+                    clusterService.submitStateUpdateTask("track-remote-cluster-metadata-changes",
+                        new ClusterStateUpdateTask() {
+
+                            @Override
+                            public ClusterState execute(ClusterState localClusterState) throws Exception {
+                                return syncMappings(subscriptionName, localClusterState, remoteClusterState);
+                            }
+
+                            @Override
+                            public void onFailure(String source, Exception e) {
+                                LOGGER.error(e);
+                            }
+                        }
+                    );
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    LOGGER.error(e);
+                }
+            });
+        }
+        cancellable = threadPool.schedule(
+            this::run,
+            this.pollDelay,
+            ThreadPool.Names.LOGICAL_REPLICATION
+        );
```

Comment:
Might make sense to defer this until after all tasks completed. Otherwise this could run into a situation where there are multiple pending cluster updates for the same subscription.